### DISCUSSION TITLE
Should every tap require Java as a dependency?

### DISCUSSION THREAD

**Discussion by helpermethod** [jreleaser/jreleaser#238] [Q&A] [answered]
Hi! 

The JReleaser Homebrew documentation says

    # A map of Homebrew dependencies.
    # The key `openjdk@${java.version}` will be added automatically if not defined.

Now, with the new BINARY distribution type, is this still a good default?

**Comment by aalmiray** [comment, marked as answer]
WE fixed https://github.com/jreleaser/jreleaser/issues/148 back in `0.4.0` which allows the Java dependency to be skipped when a distribution is of type `JAVA_BINARY` or `SINGLE_JAR`. The Java dependency is not automatically added for any other type of distribution.

There are 2 other properties named `skipBrew` and `sdkmanSkip` that perform similar behavior, i.e, skip a particular feature. I think it's better to have a consistent naming convention.

**Reply by aalmiray** [reply]
https://github.com/jreleaser/jreleaser/issues/239

**Reply by aalmiray** [reply]
Fixed and documented https://github.com/jreleaser/jreleaser.github.io/commit/a4d50d9adcb37b0715fb2c15d897636a94036ac4
